src/Build/BackEnd/Components/Communications/NodeEndpointInProc.cs

```diff
@@ -437,25 +435,11 @@ private void PacketPumpProc()
                                 // get into a condition where the sending thread continues to pump packets in
                                 // as fast as we can send them, potentially starving us from detecting a
                                 // terminate event.
-                                int packets = 0;
-                                lock (_packetQueue)
-                                {
-                                    packets = _packetQueue.Count;
-                                }
 
-                                while (packets > 0)
+                                INodePacket packet;
```

Comment:
Now that this code is simplified, you can probably delete the comment above it.  The comment almost indicates that the `while()` should be checking for the termination signal but the code isn't doing that.

Reply:
Done